src/test/java/spoon/test/template/TemplateTest.java

```diff
@@ -436,4 +436,18 @@ public void testTemplateMatcherMatchTwoSnippets() throws Exception {
 
 		assertTrue(match1.equals(match2));
 	}
+	@Test
+	public void testTemplateInvocationSubstitution() throws Exception {
+		Launcher spoon = new Launcher();
```

Comment:
contract: the template engine supports substitution of method names in method calls.